c/zis/services/secmgmttss.c

```diff
@@ -0,0 +1,570 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+#include <metal/metal.h>
+#include <metal/stddef.h>
+#include <metal/stdio.h>
+#include <metal/string.h>
+
+#ifndef METTLE
+#error Non-metal C code is not supported
+#endif
+
+#include "zowetypes.h"
+#include "alloc.h"
+#include "crossmemory.h"
+#include "radmin.h"
+#include "recovery.h"
+#include "zos.h"
+
+#include "zis/utils.h"
+#include "zis/services/secmgmt.h"
+#include "zis/services/tssparsing.h"
+
+/* A struct to more easily carry the required arguments
+ * for logging in ZIS into handler and helper functions.
+ */
+typedef struct LogData_t {
+  CrossMemoryServerGlobalArea *globalArea;
+  int traceLevel;
+} LogData;
+
+/* A helper function which finds the line where the value where we are looking for starts
+ * and returns a pointer to it. This is useful for when the caller wants to start from
+ * a particular value because they either reached the maximum they could return in a 
+ * request, or if they want to information on a single user. It also returns pointers
+ * to the lastByteOffset and the blockBytesExtracted so when continuing to parse,
+ * we still have the right place through the output.
+ */
+static const RadminVLText *findLineOfStart(const RadminVLText *currentLine, const char *searchValue, 
+                                           int searchValueMaxLength, const char *searchKey,
+                                           const unsigned int *lastByteOffset, unsigned int *blockBytesExtracted,
```

Comment:
A better name for ```blockBytesExtracted``` would be ```offsetInBlock``` and for ```lastByteOffset``` is ```maxOffsetInBlock```, that will indicate the purpose better.